core/src/main/kotlin/motif/models/parsing/ObjectsClassParser.kt

```diff
@@ -125,6 +125,13 @@ class ObjectsClassParser : ParserUtil {
 
         val returnType: IrType = method.returnType
         val returnClass: IrClass = returnType.resolveClass() ?: throw TypeNotSpreadable(objectsClass, method, returnType)
+
+        returnClass.methods
+                .find { it.annotations.any { it.type.simpleName == "Nullable" } }
+                ?.let {
+                    throw NullableSpreadMethodError(returnClass, it)
+                }
+
         val methods: List<SpreadMethod> = returnClass.methods
                 .filter { !it.isVoid() && it.isPublic() && !it.hasParameters() }
```

Comment:
This check ensures that we only consider public/non-void/parameterless methods. I'm thinking it might make more sense to only validate such methods for the nullable annotation. What do you think? - If we go that route we should also add a test that ensures that Nullable annotations are fine on methods that don't satisfy these conditions.